src/StreamJsonRpc/Exceptions/RemoteInvocationException.cs

```diff
@@ -83,5 +85,56 @@ public RemoteInvocationException(string? message, int errorCode, object? errorDa
         /// The default implementation of this method produces a <see cref="CommonErrorData"/> object.
         /// </remarks>
         public object? DeserializedErrorData { get; }
+
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            string result = base.ToString();
+
+            CommonErrorData? errorData = this.DeserializedErrorData as CommonErrorData;
+            if (errorData != null)
+            {
+                var builder = new StringBuilder(base.ToString());
```

Comment:
You can use "result" instead of "base.ToString()" here.

Reply:
This feels right to me in terms of readability. If the ToString() method affects the performance considerably then we should go with this, but I would suggest changing the name of the variable to baseString or something similar.